pkg/api/v1/clients/multicluster/multi_cluster_resource_client_kube_test.go

```diff
@@ -0,0 +1,239 @@
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/wrapper"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	multicluster2 "github.com/solo-io/solo-kit/pkg/multicluster"
+	"github.com/solo-io/solo-kit/pkg/multicluster/clustercache"
+	"github.com/solo-io/solo-kit/pkg/multicluster/secretconverter"
+	skpkg "github.com/solo-io/solo-kit/pkg/multicluster/v1"
+	"github.com/solo-io/solo-kit/test/helpers"
+	v1 "github.com/solo-io/solo-kit/test/mocks/v1"
+	"github.com/solo-io/solo-kit/test/testutils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var _ = Describe("MultiClusterResourceClient e2e test", func() {
+	if os.Getenv("RUN_KUBE_TESTS") != "1" {
+		log.Printf("This test creates kubernetes resources and is disabled by default. To enable, set RUN_KUBE_TESTS=1 in your env.")
+		return
+	}
+
+	var (
+		namespace       string
+		subject         v1.AnotherMockResourceClient
+		watchAggregator wrapper.WatchAggregator
+		localRestConfig *rest.Config
+		localKubeClient kubernetes.Interface
+		ctx             context.Context
+		cancel          context.CancelFunc
+		cfgSecret       *corev1.Secret
+		remoteCluster   = "cluster-two"
+		// resources on different clusters can have the same name and namespace
+		resourceName = "name"
+	)
+
+	getAnotherMockResource := func(cluster, ns, name, basicField string) *v1.AnotherMockResource {
+		return &v1.AnotherMockResource{
+			Metadata: core.Metadata{
+				Name:      name,
+				Namespace: ns,
+				Cluster:   cluster,
+			},
+			BasicField: basicField,
+		}
+	}
+
+	BeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.Background())
+		cacheManager, err := clustercache.NewCacheManager(ctx, kube.NewKubeSharedCacheForConfig)
+		Expect(err).NotTo(HaveOccurred())
+		clientGetter := clientgetter.NewKubeResourceClientGetter(
+			cacheManager,
+			v1.AnotherMockResourceCrd,
+			false,
+			nil,
+			0,
+			factory.NewResourceClientParams{ResourceType: &v1.AnotherMockResource{}},
+		)
+		watchAggregator = wrapper.NewWatchAggregator()
+		watchHandler := multicluster.NewAggregatedWatchClusterClientHandler(watchAggregator)
+		clientSet := multicluster.NewClusterClientManager(context.Background(), clientGetter, watchHandler)
+		mcrc := multicluster.NewMultiClusterResourceClient(&v1.AnotherMockResource{}, clientSet)
+
+		configWatcher := multicluster2.NewKubeConfigWatcher()
+		localRestConfig, err = kubeutils.GetConfig("", os.Getenv("KUBECONFIG"))
+		Expect(err).NotTo(HaveOccurred())
+		localKubeClient, err = kubernetes.NewForConfig(localRestConfig)
+		Expect(err).NotTo(HaveOccurred())
+		localCache, err := cache.NewKubeCoreCache(ctx, localKubeClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		restConfigHandler := multicluster2.NewRestConfigHandler(configWatcher, cacheManager, clientSet)
+		_, err = restConfigHandler.Run(ctx, localRestConfig, localKubeClient, localCache)
```

Comment:
having something to handle the `<-chan error` here could catch/prevent future test failures fwiw